src/main/java/uk/gov/companieshouse/extensions/api/attachments/file/FileTransferApiClient.java

```diff
@@ -118,8 +119,9 @@ public FileTransferApiClientResponse download(String fileId, HttpServletResponse
      * @param clientHttpResponse the response back from the api we are calling - the file-transfer-api
      */
     private void setResponseHeaders(HttpServletResponse httpServletResponse, ClientHttpResponse clientHttpResponse) {
-        HttpHeaders incomingHeaders = clientHttpResponse.getHeaders();
-        if (incomingHeaders != null) {
+        Optional<HttpHeaders> incomingHeadersOpt = Optional.of(clientHttpResponse.getHeaders());
```

Comment:
Sonar threw a failure on this new change, looks like the same reason as the old version.  clientHttpResponse.getHeaders() cannot return null so the fix would be just to have the previous code and remove "if (incomingHeaders != null)"

Reply:
Addressed this.